### ISSUE TITLE
This is the worst library in the react ecosystem and it's embarrassing that we let it become the standard

### ISSUE THREAD

**Issue by justrossthings** [ReactTraining/react-router#6773]
How is it acceptable for Link components to silent error and not redirect the application ? Because I used PureComponent somewhere in my component hierarchy? This is absolute trash. You say you don't connect to the redux store yet "occasionally" [your words] you fail SILENTLY when a component is not connected to store?

It's mind-blowing. ReactTraining are a true example of our universe bestowing success at random - they are the most incompetent contributors to the react ecosystem yet are paid to teach others and are trusted to handle THE MOST IMPORTANT part of a react application.

**Comment by MeiKatz** [comment]
Well, it isn't that there are no contributers willing to fix bugs and add features, it seems to me that there are to less maintainers and a leek of communication. I mean, the heads of this library are working on a new major version but in the time between we could fix bugs and add features without breaking changes that go near the new major version. Some of my pull requests are pending since months and are not merged yet. And some other pull requests are pending for years.

What's your point on this @timdorr?

**Comment by W3stside** [comment]
Got nothing to add, just here with my popcorn 🍿

**Comment by justrossthings** [comment]
@MeiKatz @timdorr _is_ the problem

v5 introduced breaking changes that @timdorr didn't feel like disclosing because "Listing out every small behavior change would be quite exhaustive and doesn't really have any impact on correctly architected React apps."

See him disregard this change here:
https://github.com/ReactTraining/react-router/issues/6671 
https://github.com/ReactTraining/react-router/issues/6657

tl;dr: 
Lead dev: introduces breaking changes without cautioning users 
Users: @$#$! you!
Lead dev:
![image](https://user-images.githubusercontent.com/8374514/59251793-9a4f8900-8c65-11e9-8dba-4dd8c7197d51.png)

**Comment by MeiKatz** [comment]
@justrossthings I am not sure about this because I am not on this boat for too long. Haven't spook with Ryan Florence yet but maybe this would be a good idea.

**Comment by timdorr** [comment]
You must be a lot of fun at parties.

**Comment by timdorr** [comment]
Also, I'm not the lead dev. That's Michael and Ryan. I'm just the janitor.

**Comment by StringEpsilon** [comment]
@justrossthings 

I do get where you're coming from, but you shouldn't immediately antagonize all maintainers by being so abrasive. 

@pshrmn @timdorr @mjackson @ryanflorence 

I think there is room for a discussion about the communication and the lack thereof.

Specifically:

* Who can make a release and how was unclear and delayed 5.0.1 unnecessarily. That particular problem is solved now, but it might be time to update the `CONTRIBUTING.md`
* A lot of people are unhappy with how their issues are closed. I see the need for the default reply to stay on top of things, but making that message a tad friendlier with an invite to clarify might help a long way to reduce the resentments. (See also #6770) [1]
* Some changes are / were poorly explained and or documented. Bringing back the changelog might help to address that particular problem.
* Communication about ongoing work is all over the place. We have the pinned timeline issue, Ryans blogpost, the Q&A on reactiflux, and  various tweets. I think this repository needs to be kept up to date. 
* Additionally, the silence on Pull-Requests that introduce new features is demotivating. A simple "Things are in flux right now and we'll take a few months to get back to you" would do a lot.

The fact that the work on 6.x (or whatever version it will be) happens completely in the dark right now is also unsatisfying.

I'd love to contribute more than just bugfixes and simple optimizations. But I can't, because i don't know what's going on with the library right now.

[1]: Something like this might work:

`
The issue tracker here on GitHub is reserved for bugreports and feature requests. 
For usage questions, please use [Stack Overflow](https://stackoverflow.com/questions/tagged/react-router) or [Reactiflux](https://www.reactiflux.com/) 
where there are a lot more people ready to help you out. 

Please feel free to clarify your issue if you think it was closed prematurely.

Thank you for your understanding.
`

**Comment by pshrmn** [comment]
@StringEpsilon those are legitimate things to discuss, but it would be better to discuss them in an issue that isn't created to insult the project and the people who work on it.

**Comment by StringEpsilon** [comment]
Should I move my comment to a new issue then?

**Comment by justrossthings** [comment]
> You must be a lot of fun at parties.

> and doesn't really have any impact on **correctly architected** React apps

The presumptuousness of this response of yours to https://github.com/ReactTraining/react-router/issues/6671 tells me everything I need to know about you.

**Comment by justrossthings** [comment]
As mentioned above, the migration v3 -> v4 was also unnecessarily painful. There's a pattern here.

**Comment by timdorr** [comment]
OK, I think we're done here...
